migration-from-4x.md

```diff
@@ -0,0 +1,328 @@
+  icon: 'insertdatetime',
+  tooltip: "Insert Current Date",
+  onclick: insertDate,
+  onpostrender: function monitorNodeChange() {
+    var btn = this;
+    editor.on('NodeChange', function(e) {
+      btn.disabled(e.element.nodeName.toLowerCase() == 'time');
+    });
+  }
+});
+```
+#####  TinyMCE 5.0:
+
+```js
+editor.ui.registry.addButton('customDateButton', {
+  icon: 'insert-time',
+  tooltip: 'Insert Current Date',
+  disabled: true,
+  onAction: (_) => editor.insertContent(toTimeHtml(new Date())),
+  onSetup: (buttonApi) => {
+    const editorEventCallback = (eventApi) => buttonApi.setDisabled(eventApi.element.nodeName.toLowerCase() === 'time');
+    editor.on('NodeChange', editorEventCallback);
+    return (buttonApi) => editor.off('NodeChange', editorEventCallback);
+  }
+});
+```
+> Note: The callback function given to `onSetup` should take a `buttonApi` argument which is an object that contains some helper functions.
+
+Each type of toolbar button has a different API. In this example, the button's API contains `isDisabled: () => boolean` and `setDisabled: (state: boolean) => void`.
+
+Also, `onpostrender` was only ever called once, when the editor is initialized. `onSetup` is called whenever the component is rendered, which may be multiple times. This means we may need a `onTearDown` handler. Rather than have to define this as a separate configuration option, `onSetup` can return a function to be used on tear down.
+
+This is particularly important if `onSetup` listens to any events using `editor.on(eventName, callback)`. Unless the event was `'init'`, `onSetup` should return `(buttonApi) => ed.off(eventName, callback)`. The **teardown** function will automatically be called by the editor when necessary.
+
+If some functionality should only run once, when the editor is initialized, it should be passed to `editor.on('init', callback)` as the callback function so it's only run when the editor is initialized.
+
+### Custom Menu Items
+
+##### New methods:
+
+| **New Method** | **Description** |
+| ----------- | -------------- |
+| editor.ui.registry.addMenuItem() | Adds a custom basic menu item. |
+| editor.ui.registry.addToggleMenuItem() | Adds a custom toggle menu item. |
+
+Docs coming soon!
+
+### Custom Dialogs
+
+Dialogs no longer have `height` or `width` settings. The dialog component now uses CSS3 and a predefined `small`, `medium`, and `large` template to specify the dimensions.
+
+### Custom Sidebars
+
+Docs coming soon!
+
+### Custom Context Toolbars
+
+The Context Toolbar takes its buttons the registry of toolbar buttons added using `addButton`, `addToggleButton`, `addSplitButton` or `addMenuButton`. The method for creating custom context toolbars has also been moved from `editor.addContextToolbar()` to `editor.ui.registry.addContextToolbar()`.
+
+For more information on Context Toolbars, see [docs]({{site.baseurl}}/components/contexttoolbar).
+
+#### Changes between TinyMCE 4.x and TinyMCE 5.0:
```

Comment:
This section mostly only lists TinyMCE 4.x behaviours, it doesn't say what's changed. @shikhanansi @EkimChau We need to chase up what the actual difference is. 

Also, in the third point what does "pop" refer to?

Reply:
We should rewrite this summary at the end, i've yet to review all the other change docs